controlplane/pkg/nsmd/remote_peer_descriptor.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2020 Cisco Systems, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nsmd
+
+import (
+	"context"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/registry"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/model"
+)
+
+//RemotePeerDescriptor represents network service manager remote peer
+type RemotePeerDescriptor interface {
+	AddConnection(connection *model.ClientConnection)
+	RemoveConnection(connection *model.ClientConnection)
+	Cancel()
+	Context() context.Context
+	Reset()
+	IsCanceled() bool
+	HasConnection() bool
+	RemoteNsm() *registry.NetworkServiceManager
+}
+
+type remotePeerDescriptor struct {
```

Comment:
Probably makes sense to make that structure thread safe

Reply:
this structure using only in locked sections. I do not think that we need to make this structure thread-safe.

Reply:
At least here you pass it inside go routine https://github.com/networkservicemesh/networkservicemesh/pull/2030/files#diff-cd75ec725c492a08d9a9a6154b4e8418R143

Reply:
there missing lock sections. Thanks, fixed.

Reply:
finally fixed, thanks